### PR TITLE
chore: upgrade tentacle, enable reuse port

### PR DESCRIPTION
**Pull request by driftluo**
1. Upgrade tentacle
2. Enable reuse port on unix, Windows system has permission problems, it cannot be turned on by default
3. The discovery protocol version number has been increased to identify whether the observed address can be broadcasted

### CODE REVIEW

**Review by doitian — DISMISSED** [review]

**Review by yangby-cryptape — APPROVED** [review]

File: network/src/protocols/discovery/mod.rs [hunk 1]
```diff
@@ -132,6 +134,9 @@ impl<M: AddressManager> ServiceProtocol for DiscoveryProtocol<M> {
                                 if let RemoteAddress::Listen(ref addr) = state.remote_addr {
                                     self.addr_mgr.add_new_addr(session.id, addr.clone());
                                 }
+                            } else if version == state::REUSE_PORT_VERSION {
```
Comment (quake) [inline network/src/protocols/discovery/mod.rs:137]: we should consider future version upgrade
```suggestion
                            } else if version >= state::REUSE_PORT_VERSION {
```

**Review by quake — CHANGES_REQUESTED** [review]

**Review by quake — DISMISSED** [review]

**Review by doitian — DISMISSED** [review]

**Comment by driftluo** [comment]
bors r=quake,doitian

**Comment by driftluo** [comment]
bors retry

**Comment by driftluo** [comment]
bors retry

**Review by doitian — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Comment by doitian** [comment]
bors r=quake,doitian

**Pull request merged**